components/channel_view/channel_view.jsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import $ from 'jquery';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import * as UserAgent from 'utils/user_agent.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+import deferComponentRender from 'components/deferComponentRender';
+
+import ChannelHeader from 'components/channel_header';
+import CreatePost from 'components/create_post.jsx';
+import FileUploadOverlay from 'components/file_upload_overlay.jsx';
+import PostView from 'components/post_view';
+import TutorialView from 'components/tutorial/tutorial_view.jsx';
+
+export default class ChannelView extends React.Component {
+    static propTypes = {
+
+        /**
+         * ID of the channel to display
+         */
+        channelId: PropTypes.string.isRequired,
+
+        /**
+         * Set if this channel is deactivated, primarily used for DMs with inactive users
+         */
+        deactivatedChannel: PropTypes.bool.isRequired,
+
+        /**
+         * Set to show the tutorial
+         */
+        showTutorial: PropTypes.bool.isRequired,
+
+        /**
+         * React router params
+         */
+        params: PropTypes.object
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.createDeferredPostView();
+    }
+
+    createDeferredPostView = () => {
+        this.deferredPostView = deferComponentRender(
+            PostView,
+            <div id='post-list'/>
+        );
+    }
+
+    componentDidMount() {
+        $('body').addClass('app__body');
+
+        // IE Detection
+        if (UserAgent.isInternetExplorer() || UserAgent.isEdge()) {
+            $('body').addClass('browser--ie');
+        }
+    }
+
+    componentWillUnmount() {
+        $('body').removeClass('app__body');
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelId !== nextProps.channelId) {
+            this.createDeferredPostView();
```

Comment:
We only want to create a new DeferredPostView if we're changing channel. This isn't too necessary since re-rendering in this component rarely happens when we're not changing channels but I figured it was better to be safe about it